packages/lwc/index.d.ts

```diff
@@ -191,42 +191,64 @@ declare module 'lwc' {
 
     /**
      * Decorator factory to wire a property or method to a wire adapter data source
+     * Use generic types to allow type checking for wire adapters
+     * Default all the generic types to any to maintain backward compatibility
```

Comment:
Is there any plan to narrow down typing in a future LWC major release?

Reply:
We could consider narrowing it down in the future if TypeScript has better support on decorators. For now I think these generic types can do the job for us and we can keep it non-breaking by using `any`.

Reply:
@pmdartus If we don't consider type changes to be breaking changes, we could just require the use of generics here. What do you think?

Reply:
@pmdartus Sorry, what I meant was that we are defaulting to `any` like so:
```
interface WireAdapterConstructor<Config extends object = any, Value = any> ...
```
So that we don't break anyone doing the following:
```
@wire(getRecord: WireAdapterConstructor)
foo;
```
If we were to simplify the above to:
```
interface WireAdapterConstructor<Config extends object, Value> ...
```
Then existing code would have to be updated:
```
@wire(getRecord: WireAdapterConstructor<GetRecordConfig, RecordRepresentation>)
foo;